pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -171,12 +171,23 @@ let
       export HOSTAR=$AR_FOR_BUILD
       export HOSTLD=$LD_FOR_BUILD
 
+      # Absolute paths for tools avoid any PATH-clobbering issues.
+      export CC=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc
+      export LD=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}ld
+      export AR=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}ar
+      export NM=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}nm
+      export STRIP=${stdenv.cc.bintools.bintools}/bin/${stdenv.cc.targetPrefix}strip
```

Comment:
I think we need to figure out what the actual problem here is, given that bintools should have a strip symlink.  It's not that it's a link to objcopy — that's on purpose, see https://github.com/NixOS/nixpkgs/pull/345691#issuecomment-2385615104.

Reply:
No. There is just no strip in `stdenv.cc.bintools`.

Could this be the reason? I have difficulty navigating this, so I might be wrong, but here the strip wrapper is only generated if it's needed:

https://github.com/NixOS/nixpkgs/blame/master/pkgs/build-support/bintools-wrapper/default.nix#L361

All the other tools that don't need a wrapper, just get symlinked here:

https://github.com/NixOS/nixpkgs/blame/master/pkgs/build-support/bintools-wrapper/default.nix#L225

Reply:
Yes. That works.